docs/en/reference/basic-mapping.rst

```diff
@@ -209,27 +209,39 @@ built-in mapping types:
 -  ``decimal``: Type that maps a SQL DECIMAL to a PHP string.
 -  ``date``: Type that maps a SQL DATETIME to a PHP DateTime
    object.
+-  ``date_immutable``: Type that maps a SQL DATETIME to a PHP DateTimeImmutable
+   object.
 -  ``time``: Type that maps a SQL TIME to a PHP DateTime object.
+-  ``time_immutable``: Type that maps a SQL TIME to a PHP DateTimeImmutable object.
 -  ``datetime``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
    DateTime object.
+-  ``datetime_immutable``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
+   DateTimeImmutable object.
 -  ``datetimetz``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
    DateTime object with timezone.
+-  ``datetimetz_immutable``: Type that maps a SQL DATETIME/TIMESTAMP to a PHP
+   DateTimeImmutable object with timezone.
```

Comment:
`DateTimeImmutable` always has a timezone. The difference is whether it's taken from the DB value or from the server configuration. Otherwise, the wording looks a bit misleading.

Reply:
So is the `datetimetz` one. Do you have a suggestion what would be more understandable for a reader? I'll change the description for both.

Reply:
- `datetime_immutable`: Type that maps a SQL DATETIME/TIMESTAMP to a PHP `DateTimeImmutable` object with the current timezone.
- `datetimetz_immutable`: Type that maps a SQL DATETIME/TIMESTAMP to a PHP `DateTimeImmutable` object with the timezone specified in the value from the database.

Reply:
Thank you.